### PR TITLE
created debug credentials provider

### PR DESCRIPTION
**Pull request by yk**
if DEBUG_FAKE_USERS is set, a credentials provider for next auth is created that lets one log in with an arbitrary username and always get accepted. this simplifies the experience for newcomers who just want to e.g. create a UI component.

### CODE REVIEW

File: website/src/components/Avatar.tsx [hunk 1]
```diff
@@ -12,7 +12,8 @@ export function Avatar() {
     return <></>;
   }
   if (session && session.user) {
-    const email = session.user.email;
+    console.log(session.user);
```
Comment (villelahdenvuo) [inline website/src/components/Avatar.tsx:15]:
```suggestion
```

Comment (yk) [reply website/src/components/Avatar.tsx]: thank you!

File: website/src/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -32,6 +34,23 @@ if (process.env.DISCORD_CLIENT_ID) {
   );
 }
 
+if (boolean(process.env.DEBUG_FAKE_USERS)) {
```
Comment (fozziethebeat) [inline website/src/pages/api/auth/[...nextauth].ts:37]: Should we just do this whenever we're running in development mode? If so we could cut out a new env variable and make this

```
if (process.env.NODE_ENV === "development") {
```

File: website/README.md [hunk 3]
```diff
@@ -63,6 +63,14 @@ If you're doing active development we suggest the following workflow:
     navigate to `http://localhost:1080`. Check the email listed and click the
     log in link. You're now logged in and authenticated.
 
+### Using debug user credentials
+
+If you want to quickly test the website, you can use the debug user credentials workflow:
```
Comment (fozziethebeat) [inline website/README.md:68]: To simplify development, I think its reasonable to always activate this.  What about putting it into the checked in `.env` file?

**Review by fozziethebeat — APPROVED** [review]
I like this, I didn't notice that provider.

I proposed two small alternatives to smooth out the setup process.

File: website/src/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (yk) [reply website/src/pages/api/auth/[...nextauth].ts]: that's a great idea. I've pushed a change for this and adjusted documentation

**Review by fozziethebeat — APPROVED** [review]

File: website/src/pages/api/auth/[...nextauth].ts [hunk 4]
```diff
@@ -32,6 +34,23 @@ if (process.env.DISCORD_CLIENT_ID) {
   );
 }
 
+if (boolean(process.env.NODE_ENV === "development")) {
```
Comment (villelahdenvuo) [inline website/src/pages/api/auth/[...nextauth].ts:37]:
```suggestion
if (process.env.NODE_ENV === "development") {
```

We can remove the dependency now.

**Pull request merged**